src/fides/api/util/logger_context_utils.py

```diff
@@ -0,0 +1,93 @@
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from requests import PreparedRequest, Response
+from requests.exceptions import (  # pylint: disable=redefined-builtin
+    ConnectionError,
+    ConnectTimeout,
+    ReadTimeout,
+    SSLError,
+    TooManyRedirects,
+)
+
+if TYPE_CHECKING:
+    from fides.api.service.connectors.saas_connector import SaaSConnector
+
+from fides.config import CONFIG
+
+
+class ErrorGroup(Enum):
+    """A collection of user-friendly error labels to be used in contextualized logs."""
+
+    network_error = "Network/protocol error"
+    authentication_error = "Authentication error"
+    client_error = "Client-side error"
+    server_error = "Server-side error"
+
+
+def saas_connector_details(connector: "SaaSConnector") -> Dict[str, Any]:
+    """Maps the system and connection info details. Includes the collection and privacy request ID if available."""
+
+    details = {
+        "system_id": (
+            connector.configuration.system.id
+            if connector.configuration.system
+            else None
+        ),
+        "connection_key": connector.configuration.key,
+    }
+    if connector.current_collection_name:
+        details["collection"] = connector.current_collection_name
+    if connector.current_privacy_request:
+        details["privacy_request_id"] = connector.current_privacy_request.id
+    return details
+
+
+def request_details(
+    prepared_request: PreparedRequest,
+    response: Optional[Response] = None,
+    ignore_error: bool = False,
+) -> Dict[str, Any]:
+    """Maps the request details and includes response details when "dev mode" is enabled."""
+
+    details = {
+        "method": prepared_request.method,
+        "url": prepared_request.url,
+    }
+    if CONFIG.dev_mode and prepared_request.body is not None:
+        details["body"] = prepared_request.body
+
+    if response is not None:
+        if CONFIG.dev_mode and response.content:
+            details["response"] = response.content.decode("utf-8")
+
+        details["status_code"] = response.status_code
+
+        # assign error group only if error should not be ignored
+        if not ignore_error:
+            if response.status_code in [401, 403]:
+                details["error_group"] = ErrorGroup.authentication_error.value
+            elif 400 <= response.status_code < 500:
+                details["error_group"] = ErrorGroup.client_error.value
+            elif 500 <= response.status_code:
+                details["error_group"] = ErrorGroup.server_error.value
+    return details
+
+
+def exception_details(exception: Exception, url: str) -> Dict[str, Any]:
+    """Maps select connection exceptions to user-friendly error details."""
+
+    details = {"error_group": ErrorGroup.network_error.value}
```

Comment:
this confuses me a bit, when considered alongside the function signature. the function seems to apply to any `Exception` generally, but here we effectively assume that it's a `network_error`. that may be the only context in which this helper function is called _currently_, but seems like something that one could easily be tripped up on when trying to extend these utilities to other use cases.

so, either i'm misunderstanding something, or i'd recommend adjusting the function name or logic (or at least docstring)